### PR TITLE
[DEV-1939] node/generic: add lookup target node

### PR DESCRIPTION
**Pull request by jevonyeoh**
## Description
Reuse the lookup node's logic, and created a new lookup target node.

## Related Issue
https://featurebyte.atlassian.net/browse/DEV-1939

## Type of Change

Label this pull request to place it under the correct category in Release Notes:

|        **Pull Request Label**         | **Category in Release Notes** |
|:-------------------------------------:|:-----------------------------:|
|       `enhancement`, `feature`        |          🚀 Features          |
| `bug`, `refactoring`, `bugfix`, `fix` |    🔧 Fixes & Refactoring     |
|       `build`, `ci`, `testing`        |    📦 Build System & CI/CD    |
|              `breaking`               |      💥 Breaking Changes      |
|            `documentation`            |       📝 Documentation        |
|            `dependencies`             |    ⬆️ Dependencies updates    |

## Checklist

- [ ] I have read the [`CODE_OF_CONDUCT.md`](https://github.com/featurebyte/featurebyte/blob/main/CODE_OF_CONDUCT.md) and [`CONTRIBUTING.md`](https://github.com/featurebyte/featurebyte/blob/main/CONTRIBUTING.md) guides.
- [ ] I have written tests for the changes made.
- [ ] I have written docstrings in [NumpyDoc format](https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard)
- [ ] I have labeled my Pull Request correctly

### CODE REVIEW

File: featurebyte/query_graph/node/generic.py [hunk 1]
```diff
@@ -924,6 +930,12 @@ def _derive_sdk_code(
         return statements, out_var_name
 
 
+class LookupTargetNode(BaseLookupNode):
+    """LookupTargetNode class"""
+
+    type: Literal[NodeType.LOOKUP_TARGET] = Field(NodeType.LOOKUP_TARGET, const=True)
+
```
Comment (gee-senbong) [inline featurebyte/query_graph/node/generic.py:937]: Curious what is the difference between `LookupNode` and `LookupTargetNode`? Could we reuse the same node if there is no difference?

File: featurebyte/query_graph/node/generic.py [hunk 2]
```diff
@@ -924,6 +930,12 @@ def _derive_sdk_code(
         return statements, out_var_name
 
 
+class LookupTargetNode(BaseLookupNode):
```
Comment (YS-L) [inline featurebyte/query_graph/node/generic.py:933]: Should this have a field for the point in time offset parameter or is the plan to add that later?

Comment (jevonyeoh) [reply featurebyte/query_graph/node/generic.py:933]: hm, yea i can add that now!

File: featurebyte/query_graph/node/generic.py (hunk 1 above)
Comment (jevonyeoh) [reply featurebyte/query_graph/node/generic.py:944]: there'll be some different parameters, and also some different handling, which requires a different node!

**Review by YS-L — APPROVED** [review]

File: featurebyte/query_graph/node/generic.py (hunk 1 above)
Comment (gee-senbong) [reply featurebyte/query_graph/node/generic.py:944]: Could we update existing lookup related tests to parametrize it to handle different lookup node type?

Comment (jevonyeoh) [reply featurebyte/query_graph/node/generic.py:944]: yeap! will update the tests accordingly :)

**Pull request merged**